### DISCUSSION TITLE
Rewrite the package with typescript, or at least publish type definitions

### DISCUSSION THREAD

**Discussion by thejessewinton** [vimeo/player.js#901] [Ideas]
I'm working with the API in a project, and the lack of types and type definitions have made it a lot less developer friendly than I would have hoped. Are there any plans to add typescript to the project at any point?

**Comment by bdougherty** [comment]
@thejessewinton there are no concrete plans at the moment, but we have it on the roadmap this year to give this library some much-needed love. I can't promise anything, but I would say that it is very likely that TypeScript will be a part of that (at least definitions).

**Comment by vuminhluan** [comment]
Hi, is there any update or progress on this?
